### ISSUE TITLE
Any option for Outline?

### ISSUE THREAD

**Issue by huytd** [neoclide/coc-denite#5]
For now, we only have `coc-symbols`, how about outline, it would be great to replace `CocList outline` with denite!

**Comment by chemzqm** [comment]
PR welcome

**Comment by huytd** [comment]
Yes, I'm trying to implement it.

From what I see, outline is a bit tricker than symbols, since it invoke a call to ctags and some other processing. And it seems like, there's no way to connect to coc.nvim's Outline function, so, we have to reimplement it in here.

Is that the right approach, or I'm missing something? what do you think @chemzqm ?

**Comment by chemzqm** [comment]
There is `languages.getDocumentSymbol` function exposed, but I think you can just use denite's outline source.

**Comment by huytd** [comment]
Ah, ok. My bad, I tried denite's outline but wasn't happy with it (the default setting) because it required to have a tags file generated at first. Coc.nvim generate the tags on the fly just for the selected file.

I copied the ctags option of coc.nvim and now it seems to works.

For those who come to this ticket in the future, as chemzqm said, we can just use Denite's outline source, with a bit modification:

```
call denite#custom#var('outline', 'command', ['ctags'])
call denite#custom#var('outline', 'options', ['-f -', '--excmd=number'])
```

The `-f -` option tells ctags to output the result to stdout instead of a tag file, so Denite can pick it up.
